R/rm.ridge.R

```diff
@@ -1,31 +1,52 @@
+#' @description
+#' Computes retention time intervals of a selected retention time range.
+#' @param rt full retention time vector.
+#' @param rt_sel Selected retention time range vector.
+#' @return A list object:
+#' \itemize{
+#'   \item over_rt - upper retention time interval
+#'   \item under_rt - lower retention time interval
+#'   \item within_rt - intermediate retention time interval
+#' }
+#' @export
+compute_rt_intervals <- function(rt, rt_sel){
+    rt_max <- max(rt[rt_sel])
+    rt_min <- min(rt[rt_sel])
+
+    over_rt <- which(rt > rt_max)
+    under_rt <-  which(rt < rt_min)
+    within_rt <- which(between(rt, rt_min, rt_max))
+
+    list(over_rt = over_rt, under_rt = under_rt, within_rt = within_rt)
+}
+
 #' Removing long ridges at the same m/z.
 #' 
 #' @description
-#' This is an internal function. It substracts a background estimated through kernel smoothing when an EIC continuously 
-#' span more than half the retention time range.
-#' @param x Retention time vector.
-#' @param y2 Intensity vector.
+#' This is an internal function. It substracts a background when an EIC continuously 
+#' span more than half the retention time range. The background is estimated through kernel smoothing.
+#' @param rt Retention time vector.
+#' @param intensity Intensity vector.
 #' @param bw Bandwidth for the kernel smoother. A very wide one is used here.
-#' @return A vector of intensity value is returned.
+#' @return A vector of intensity values at each rt intervals is returned.
 #' @importFrom dplyr between
 #' @export
-rm.ridge <- function(x,y2, bw) {
-    sel<-which(y2<quantile(y2, 0.75))
-    max.x.sel<-max(x[sel])
-    min.x.sel<-min(x[sel])
-    
-    in.sel<-which(between(x, min.x.sel, max.x.sel))
-    over.sel<-which(x>max.x.sel)
-    under.sel<-which(x<min.x.sel)
-    
-    
-    this.s<-ksmooth(x[sel],y2[sel],x.points=x[in.sel],kernel="normal",bandwidth=bw)
-    if(sum(is.na(this.s$y))>0) return(y2)
+rm.ridge <- function(rt, intensity, bw) {
+    this_rt <- which(intensity < quantile(intensity, 0.75))
+
+    rt_intervals <- compute_rt_intervals(rt, this_rt)
+
+    rt_over <- rt_intervals$over_rt 
+    rt_under <- rt_intervals$under_rt
+    rt_within <- rt_intervals$within_rt
+
+    this.s <- ksmooth(rt[this_rt], intensity[this_rt], x.points = rt[rt_within], kernel = "normal", bandwidth = bw)
+    if(sum(is.na(this.s$y)) > 0) return(intensity)
```

Comment:
It may be better to reformat this one-line conditional into two lines with proper syntax. So that it's clearly visible that `rm.ridge` can finish here.

Reply:
Done.